process/sync/baseForkDetector.go

```diff
@@ -265,7 +265,7 @@ func (bfd *baseForkDetector) append(hdrInfo *headerInfo) bool {
 	defer bfd.mutHeaders.Unlock()
 
 	hdrInfos := bfd.headers[hdrInfo.nonce]
-	isHdrInfosNilOrEmpty := hdrInfos == nil || len(hdrInfos) == 0
+	isHdrInfosNilOrEmpty := len(hdrInfos) == 0 // no need for nil check, len() for nil returns 0
```

Comment:
remove variable and just check if len(hdrInfos) == 0...

Reply:
i'd leave it like this because it's easier to understand